src/services/transaction.ts

```diff
@@ -0,0 +1,35 @@
+import { Deferrable } from "@ethersproject/properties";
+import { TransactionRequest, TransactionResponse } from "@ethersproject/providers";
+
+import { ChainId } from "../chain";
+import { Service } from "../common";
+import { Context } from "../context";
+import { SdkError } from "../types/common";
+import { AllowListService } from "./allowlist";
+
+export class TransactionService<T extends ChainId> extends Service {
+  private allowListService?: AllowListService<T>;
+
+  constructor(chainId: T, ctx: Context, allowListService?: AllowListService<T>) {
+    super(chainId, ctx);
+    this.allowListService = allowListService;
+  }
+
+  async sendTransaction(transaction: Deferrable<TransactionRequest>): Promise<TransactionResponse> {
+    const valid = await this.validateTx(transaction);
+    if (!valid) {
+      throw new SdkError("transaction is not valid");
```

Comment:
why are we not using the error that comes from `validateCalldata` method? We are parsing every possible error as it is a transaction being invalid, but it could not be the case since for example `validateCalldata` call to the SC could fail for other reasons.

Reply:
good point, done